code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/AEPPushTemplate.kt

```diff
@@ -0,0 +1,400 @@
+        }
+        this.smallIcon = smallIcon
+        largeIcon =
+            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.LARGE_ICON, null)
+        try {
+            val count = data[PushTemplateConstants.PushPayloadKeys.BADGE_NUMBER]
+            count?.let {
+                badgeCount = count.toInt()
+            }
+        } catch (e: NumberFormatException) {
+            Log.debug(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Exception in converting notification badge count to int - ${e.localizedMessage}."
+            )
+        }
+        notificationPriority = NotificationPriority.from(
+            data[PushTemplateConstants.PushPayloadKeys.NOTIFICATION_PRIORITY]
+        )
+        notificationImportance =
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) getNotificationImportanceFromString(
+                data[PushTemplateConstants.PushPayloadKeys.NOTIFICATION_PRIORITY]
+            ) else null
+
+        notificationVisibility = getNotificationVisibilityFromString(
+            data[PushTemplateConstants.PushPayloadKeys.NOTIFICATION_VISIBILITY]
+        )
+        channelId =
+            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.CHANNEL_ID, null)
+        templateType =
+            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.TEMPLATE_TYPE, null)
+                ?.let { PushTemplateType.fromString(it) }
+        tag = DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.TAG, null)
+        val stickyValue =
+            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.STICKY, null)
+        isNotificationSticky =
+            if (StringUtils.isNullOrEmpty(stickyValue)) false else java.lang.Boolean.parseBoolean(
+                stickyValue
+            )
+        ticker = DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.TICKER, null)
+        expandedBodyText = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.EXPANDED_BODY_TEXT, null
+        )
+        expandedBodyTextColor = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.EXPANDED_BODY_TEXT_COLOR, null
+        )
+        titleTextColor = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.TITLE_TEXT_COLOR, null
+        )
+        smallIconColor = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.SMALL_ICON_COLOR, null
+        )
+        notificationBackgroundColor = DataReader.optString(
+            data,
+            PushTemplateConstants.PushPayloadKeys.NOTIFICATION_BACKGROUND_COLOR,
+            null
+        )
+        isFromIntent = false
+    }
+
+    /**
+     * Constructor to create a push template object from an [Intent].
+     *
+     * @param intent [Intent] containing key value pairs extracted from a push template data payload
+     */
+    constructor(intent: Intent?) {
```

Comment:
Can we have the caller do the null check on the intent and change this constructor to accept a no-null intent.